esp-hal/src/parl_io.rs

```diff
@@ -1114,13 +1141,15 @@ where
     where
         CH: DmaChannelConvert<<PARL_IO as DmaEligible>::Dma>,
     {
+        let guard = PeripheralGuard::new(crate::system::Peripheral::ParlIo);
         internal_init(frequency)?;
 
         Ok(Self {
             tx: TxCreator {
                 tx_channel: dma_channel.tx.degrade(),
                 descriptors,
             },
+            _guard: guard,
```

Comment:
I think this needs to be in the TxCreator, otherwise we can move `tx` out, but drop the peripheral.